tls/s2n_psk.c

```diff
@@ -99,6 +99,25 @@ S2N_RESULT s2n_psk_clone(struct s2n_psk *new_psk, struct s2n_psk *original_psk)
     return S2N_RESULT_OK;
 }
 
+S2N_RESULT s2n_resumption_psks_remove(struct s2n_connection *conn)
+{
+    RESULT_ENSURE_REF(conn);
+
+    struct s2n_array *psk_list = &conn->psk_params.psk_list;
+    for (uint32_t i = psk_list->len; i > 0; i--) {
+        /* This index variable along with the loop condition keep size_t i from underflowing */
+        size_t idx = i - 1;
```

Comment:
We could just start the loop from `i = psk_list->len - 1` and use `i` instead of using another `idx` here.

Reply:
We do need the idx variable here. When i = 1 in this loop, idx can safely = 0 and check the first element in the list. If we remove idx, the loop will stop before we check the first element in the list (because of the loop condition i > 0). And if we change the loop condition to i >=0, i will underflow when i = 0.

Reply:
s2n_array_get takes in uint32_t type for index, so why will it underflow when uint32_t  i = 0 here?

Reply:
Gotcha, Infinite loop problem here :) We could just prevent this by not doing backward loop traversal, we are removing all the elements in the list as it is!